narwhals/_utils.py

```diff
@@ -1671,6 +1672,12 @@ def is_compliant_expr(
     return hasattr(obj, "__narwhals_expr__")
 
 
+def is_namespace_accessor(
+    obj: NamespaceAccessor[_FullContext] | Any,
+) -> TypeIs[NamespaceAccessor[_FullContext]]:
+    return _hasattr_static(obj, "compliant")
```

Comment:
Shouldn't we better check for the `_accessor` attribute (also considering the function name)?

Reply:
@dangotbanned I did a commit for it: https://github.com/narwhals-dev/narwhals/pull/3127/commits/88e9e9fa4f75c9f0afbd5ad49a4ce4181f031521

I hope you agree with it?! 🤞🏼